pkgs/by-name/bi/bitwarden-directory-connector/package.nix

```
@@ -0,0 +1,66 @@
+{
+  lib,
+  buildNpmPackage,
+  fetchFromGitHub,
+  pkgs,
+  git,
+  python3,
+  pkg-config,
+  libsecret,
+  nodejs_18,
+  makeWrapper,
+}: let
+  buildNpmPackage' = buildNpmPackage.override {nodejs = nodejs_18;};
+in
+  buildNpmPackage' rec {
+    pname = "bitwarden-directory-connector";
+    version = "2023.10.0";
+
+    src = fetchFromGitHub {
+      owner = "bitwarden";
+      repo = "directory-connector";
+      rev = "v${version}";
+      hash = "sha256-CgaCnMWNVWCJBypNcdoseVCwD8Mlq4YaWpK+VZT/7Qk=";
+      leaveDotGit = true;
+    };
+
```

`leaveDotGit` is very hash-fragile. It's an absolute last resort. In fact, I can no longer build your version because the `src` hash doesn't match. This works for me, and avoids that problem:
```suggestion
      hash = "sha256-PlOtTh+rpTxAv8ajHBDHZuL7yeeLVpbAfKEDPQlejIg=";
     };
 
    postPatch = ''
      jq 'del(.scripts.preinstall)' package.json >package.json.tmp
      mv -f package.json{.tmp,}
    '';
```
Don't forget to also replace `git` with `jq` in the arguments and `nativeBuildInputs`.

---

Those preinstall scripts caused me no end of issues, didnt realise ye could just delete them.

---

That works, so glad I dont have to care about teh preinstall scripts :D

---

Done